code/game/machinery/doors/airlock.dm

```diff
@@ -977,6 +1012,32 @@ There are 9 wires.
 		hit(user, I)
 		return
 
+	if(istype(I, /obj/item/weapon/keys))
+		if(used_now)
+			to_chat(user, SPAN_WARNING("You are already looking for the key!")) //don't want people stacking odds
+			return
+		used_now = TRUE
+		if(ishuman(usr))
+			var/mob/living/carbon/human/H = usr
+			if(istype(I, /obj/item/weapon/keys/lockpicks))
+				playsound(src.loc, 'sound/items/keychainrattle.ogg', 30, 1, -2)
+			else
+				playsound(src.loc, 'sound/items/keychainrattle.ogg', 700, 1, -2)
+			if(do_after(user, 900, src))
+				used_now = FALSE
+				if(locked)
```

Comment:
It should check if the door is locked before it starts the timer, earlier return failure should always be first.

Reply:
It should also check if the door has power or not.

Reply:
If it has no power then it dosnt open as its not forced!
Also its a 100% intented to not have it check before the bolt as you can hide away bolt light giving janitors a soft-check rather then a hard check